### ISSUE TITLE
Riak-debug fails on stock Solaris 10 [JIRA: RIAK-2314]

### ISSUE THREAD

**Issue by kesslerm** [basho/riak#796]
On behalf of a customer:

riak-debug on solaris 10 isn't working...

```
riak-node:/opt/riak/etc root$ /opt/riak/bin/riak-debug 
..............................E....Eawk: syntax error near line 1 
awk: illegal statement near line 1 
sed: Extra text at end of command: s/^"//" -e "s/" /:/ 
Esed: Extra text at end of command: s/^"//" -e "s/".*$// 
/opt/riak/bin/riak-debug[507]: test: argument expected 
/riak_core_ring* not found 
E.Eawk: syntax error near line 1 
awk: illegal statement near line 1 
sed: Extra text at end of command: s/^"//" -e "s/".*$// 
/opt/riak/bin/riak-debug[540]: test: argument expected 
sed: Extra text at end of command: s/^"//" -e "s/".*$// 
/opt/riak/bin/riak-debug[546]: test: argument expected 
awk: syntax error near line 1 
awk: illegal statement near line 1 
sed: Extra text at end of command: s/^"//" -e "s/".*$// 
/opt/riak/bin/riak-debug[614]: test: argument expected 
.sed: Extra text at end of command: s/^"//" -e "s/".*$// 
sed: Extra text at end of command: s/^"//" -e "s/".*$// 
sed: Extra text at end of command: s/^"//" -e "s/".*$// 
/opt/riak/bin/riak-debug[794]: test: argument expected 
./tmp/riak@node-riak-debug: bad substitution 
/tmp/riak@node-riak-debug/config: bad substitution 
/tmp/riak@node-riak-debug/config: bad substitution 
/tmp/riak@node-riak-debug/config: bad substitution 
/tmp/riak@node-riak-debug/config: bad substitution 
/tmp/riak@node-riak-debug/config: bad substitution 
/tmp/riak@node-riak-debug/config: bad substitution 
/tmp/riak@node-riak-debug/config: bad substitution 
/tmp/riak@node-riak-debug/config: bad substitution 
/tmp/riak@node-riak-debug/config: bad substitution 
/tmp/riak@node-riak-debug/config: bad substitution 
/tmp/riak@node-riak-debug/config: bad substitution 
/tmp/riak@node-riak-debug/config: bad substitution 
.tar: z: unknown function modifier 
Usage: tar {c|r|t|u|x}[BDeEFhilmnopPqTvw@[0-7]][bfk][X...] [blocksize] [tarfile] [size] [exclude-file...] {file | -I include-file | -C directory file}... 
/opt/riak/etc/riakprod@riak-node-riak-debug.tar.gz 
```

The customer reported against Riak 2.0.6 but we see the same behaviour still on 2.1.2.

**Comment by kesslerm** [comment]
The following patch against riak-debug from 2.1.2 seems to work on stock Solaris 10

```
--- /opt/riak/sbin/riak-debug.orig  Wed Dec  2 00:25:23 2015
+++ /opt/riak/sbin/riak-debug   Thu Dec  3 17:51:49 2015
@@ -30,7 +30,8 @@
     POSIX_SHELL="true"
     export POSIX_SHELL
     # To support 'whoami' add /usr/ucb to path
-    PATH=/usr/ucb:$PATH
+    # To use 'nawk' as 'awk', add /usr/xpg4/bin to path
+    PATH=/usr/xpg4/bin:/usr/ucb:$PATH
     export PATH
     exec /usr/bin/ksh $0 "$@"
 fi
@@ -491,7 +492,7 @@
     riak_epaths=`make_app_epaths "${riak_app_config}"`

     # Get one http listener (epath might output a newline-separated list).
-    riak_api_http="`epath 'riak_api http' "$riak_epaths" | sed -e 's/^"//' -e 's/" /:/' | head -n1`"
+    riak_api_http="`epath 'riak_api http' "$riak_epaths" | sed -e 's/^\"//' -e 's/\" /:/' | head -n1`"

     # Dump the output of the /stats HTTP endpoint.
     dump riak_http_stats curl -s "http://$riak_api_http/stats"
@@ -500,7 +501,7 @@
     mkdir_or_die "${start_dir}"/"${debug_dir}"/ring/.info
     cd "${start_dir}"/"${debug_dir}"/ring

-    ring_dir="`epath 'riak_core ring_state_dir' "$riak_epaths" | sed -e 's/^"//' -e 's/".*$//'`"
+    ring_dir="`epath 'riak_core ring_state_dir' "$riak_epaths" | sed -e 's/^\"//' -e 's/\".*$//'`"
     if [ '/' != `echo "$ring_dir" | cut -c1` ]; then
         # relative path. prepend base dir
         ring_dir="$riak_base_dir"/"$ring_dir"
@@ -533,13 +534,13 @@
     # if not already made, make a flat, searchable version of the app.config
     [ -z "$riak_epaths" ] && riak_epaths=`make_app_epaths "${riak_app_config}"`

-    yz_dir="`epath 'yokozuna root_dir' "$riak_epaths" | sed -e 's/^"//' -e 's/".*$//'`"
+    yz_dir="`epath 'yokozuna root_dir' "$riak_epaths" | sed -e 's/^\"//' -e 's/\".*$//'`"
     if [ '/' != `echo "$yz_dir" | cut -c1` ]; then
         # relative path. prepend base dir
         yz_dir="$riak_base_dir"/"$yz_dir"
     fi

-    yz_aae_dir="`epath 'yokozuna anti_entropy_data_dir' "$riak_epaths" | sed -e 's/^"//' -e 's/".*$//'`"
+    yz_aae_dir="`epath 'yokozuna anti_entropy_data_dir' "$riak_epaths" | sed -e 's/^\"//' -e 's/\".*$//'`"
     if [ '/' != `echo "$yz_aae_dir" | cut -c1` ]; then
         # relative path. prepend base dir
         yz_aae_dir="$riak_base_dir"/"$yz_aae_dir"
@@ -607,7 +608,7 @@
     # if not already made, make a flat, searchable version of the app.config
     [ -z "$riak_epaths" ] && riak_epaths=`make_app_epaths "${riak_app_config}"`

-    log_dir="`epath 'riak_core platform_log_dir' "$riak_epaths" | sed -e 's/^"//' -e 's/".*$//'`"
+    log_dir="`epath 'riak_core platform_log_dir' "$riak_epaths" | sed -e 's/^\"//' -e 's/\".*$//'`"
     if [ '/' != `echo "$log_dir" | cut -c1` ]; then
         # relative path. prepend base dir
         log_dir="$riak_base_dir"/"$log_dir"
@@ -633,9 +634,9 @@
     fi

     # Lager info and error files
-    new_format_lager_files="`epath 'lager handlers lager_file_backend file' "$riak_epaths" | sed -e 's/^"//' -e 's/".*$//'`"
+    new_format_lager_files="`epath 'lager handlers lager_file_backend file' "$riak_epaths" | sed -e 's/^\"//' -e 's/\".*$//'`"
     if [ -z "$new_format_lager_files" ]; then
-        lager_files="`epath 'lager handlers lager_file_backend' "$riak_epaths" | cut -d' ' -f 1 | sed -e 's/^"//' -e 's/".*$//'`"
+        lager_files="`epath 'lager handlers lager_file_backend' "$riak_epaths" | cut -d' ' -f 1 | sed -e 's/^\"//' -e 's/\".*$//'`"
     else
         lager_files=$new_format_lager_files
     fi
@@ -662,7 +663,7 @@
     # Gather backend logs, listing, sizing information, etc..
     backend=`epath 'riak_kv storage_backend' "$riak_epaths"`
     if [ 'riak_kv_eleveldb_backend' = "$backend" ]; then
-        leveldb_dir="`epath 'eleveldb data_root' "$riak_epaths" | sed -e 's/^"//' -e 's/".*$//'`"
+        leveldb_dir="`epath 'eleveldb data_root' "$riak_epaths" | sed -e 's/^\"//' -e 's/\".*$//'`"

         if [ '/' != `echo "$leveldb_dir" | cut -c1` ]; then
             # relative path. prepend base dir
@@ -693,7 +694,7 @@
             ' "${start_dir}"/"${debug_dir}"/logs/leveldb {} \;

     elif [ 'riak_kv_bitcask_backend' = "$backend" ]; then
-        bitcask_dir="`epath 'bitcask data_root' "$riak_epaths" | sed -e 's/^"//' -e 's/".*$//'`"
+        bitcask_dir="`epath 'bitcask data_root' "$riak_epaths" | sed -e 's/^\"//' -e 's/\".*$//'`"

         if [ '/' != `echo "$bitcask_dir" | cut -c1` ]; then
             # relative path. prepend base dir
@@ -723,7 +724,7 @@
             backend="`epath "riak_kv multi_backend $b" "$riak_epaths" | cut -d ' ' -f 1 | uniq`"
             if [ 'riak_kv_eleveldb_backend' = "$backend" ]; then
                 dr="`epath "riak_kv multi_backend $b riak_kv_eleveldb_backend data_root" "$riak_epaths" |
-                    sed -e 's/^"//' -e 's/".*$//'`"
+                    sed -e 's/^\"//' -e 's/\".*$//'`"

                 if [ '/' != `echo "$dr" | cut -c1` ]; then
                     # relative path. prepend base dir
@@ -755,7 +756,7 @@

             elif [ 'riak_kv_bitcask_backend' = "$backend" ]; then
                 dr="`epath "riak_kv multi_backend $b riak_kv_bitcask_backend data_root" "$riak_epaths" |
-                    sed -e 's/^"//' -e 's/".*$//'`"
+                    sed -e 's/^\"//' -e 's/\".*$//'`"

                 if [ '/' != `echo "$dr" | cut -c1` ]; then
                     # relative path. prepend base dir
@@ -786,7 +787,7 @@
     # an empty variable regardless of the value set in the configuration
     # settings. We're going to grab the `anti_entropy_data_dir` instead, and
     # assume it's presence on disk indicates activity.
-    anti_entropy_dir="`epath 'riak_kv anti_entropy_data_dir' "$riak_epaths" | sed -e 's/^"//' -e 's/".*$//'`"
+    anti_entropy_dir="`epath 'riak_kv anti_entropy_data_dir' "$riak_epaths" | sed -e 's/^\"//' -e 's/\".*$//'`"

     if [ '/' != `echo "$anti_entropy_dir" | cut -c1` ]; then
         # relative path. prepend base dir
@@ -899,9 +900,9 @@

 if [ '-' = "$outfile" ]; then
     # So we don't get a file literally named -
-    tar zcf - "${debug_dir}"
+    tar cf - "${debug_dir}" | gzip
 else
-    tar zcf "$outfile" "${debug_dir}"
+    tar cf - "${debug_dir}" | gzip > "$outfile"

     # provide some indication of the output filename
     printf " $outfile" 1>&2
```

**Comment by Basho-JIRA** [comment]
Needs to forward merge 

**_[posted via JIRA by Patricia Brewer]**_

**Comment by Basho-JIRA** [comment]
https://github.com/basho/riak/pull/798 for riak

https://github.com/basho/riak_ee/pull/365 for riak_ee

Both are in the 2.0 branches of the respective projects - just need to be forward-merged to 2.1 (and make sure we don't pick up anything else unless it's relevant). 

**_[posted via JIRA by Douglas Rohrer]**_

**Comment by Basho-JIRA** [comment]
https://github.com/basho/riak/pull/808 for riak - riak_ee was the first commit on the develop-2.2 branch w/o PR. 

**_[posted via JIRA by Douglas Rohrer]**_

**Comment by Basho-JIRA** [comment]
Confirmed fix is on riak and riak_ee, both for 2.0 and develop-2.2 branches 

**_[posted via JIRA by Douglas Rohrer]**_
